packages/arcs/src/arc_link_labels/ArcLinkLabelsLayer.tsx

```diff
@@ -56,46 +67,149 @@ export const ArcLinkLabelsLayer = <Datum extends DatumWithArcAndColor>({
+    activeId?: string | number
+    component: any
+    label: ArcLinkLabelsProps<Datum>['arcLinkLabel']
+    strokeWidth: number
+    transitionProps: any
+    datum: Datum
+    interpolateLink: any
+    interpolateTextAnchor: any
+    interpolateTextPosition: any
+}
+
+function ArcLinkLabelItem<Datum extends DatumWithArcAndColor>({
+    clip,
+    setClip,
+    activeId,
+    component,
+    label,
+    strokeWidth,
+    transitionProps,
+    datum,
+    interpolateLink,
+    interpolateTextAnchor,
+    interpolateTextPosition,
+}: ArcLinkLabelItemProps<Datum>) {
+    const getLabel = usePropertyAccessor<Datum, string>(label)
+    const Label: ArcLinkLabelComponent<Datum> = component
+    const dimensions = clip.dimensions[datum.id]
+
+    const ref = useCallback((domNode: SVGElement | null) => {
+        if (domNode) {
+            // the intention is to gather dimensions of all labels then update the state of the parent 
+            // to re-render once all the dimensions are gathered
+            const hasDims = !!clip.dimensions[datum.id]
+            clip.dimensions[datum.id] = domNode.getBoundingClientRect()
+            if (!hasDims) {
+                --clip.dimsRequired
+                if (clip.dimsRequired === 0) setClip({ ...clip, avoid: [] })
+            }
+        }
+    }, [setClip, activeId])
+
+    let clipped = true
+
+    if (dimensions && clip.avoid) {
+        if (datum.id == activeId ||
+            !clip.avoid.reduce((found: boolean, rect: DOMRect) => found || intersects(dimensions, rect), false)) {
+            clipped = false
+            clip.avoid.push(dimensions)
+        }
+    } else if (!dimensions) {
+        delete clip.dimensions[datum.id]
+    }
+
+    const opacity = clipped ? 0.05 : (activeId && datum.id !== activeId ? 0.5 : 1)
+    const style = {
+        ...transitionProps,
+        opacity,
+        thickness: strokeWidth,
+        path: interpolateLink(
+            transitionProps.startAngle,
+            transitionProps.endAngle,
+            transitionProps.innerRadius,
+            transitionProps.outerRadius,
+            transitionProps.offset,
+            transitionProps.diagonalLength,
+            transitionProps.straightLength
+        ),
+        textAnchor: interpolateTextAnchor(
+            transitionProps.startAngle,
+            transitionProps.endAngle,
+            transitionProps.innerRadius,
+            transitionProps.outerRadius
+        ),
+        textPosition: interpolateTextPosition(
+            transitionProps.startAngle,
+            transitionProps.endAngle,
+            transitionProps.innerRadius,
+            transitionProps.outerRadius,
+            transitionProps.offset,
+            transitionProps.diagonalLength,
+            transitionProps.straightLength,
+            transitionProps.textOffset
+        )
+    }
+
+    return <g ref={ref} key={datum.id}>{React.createElement(Label, { datum, label: getLabel(datum), style })}</g>
```

Comment:
The SVG Pie allows a function to create the arc label to be injected, so we have to wrap whatever it creates with a group to get the bounds